mne/preprocessing/infomax_.py

```diff
@@ -136,7 +148,8 @@ def infomax(data, weights=None, l_rate=None, block=None, w_change=1e-12,
     while step < max_iter:
 
         # shuffle data at each step
-        rng.seed(step)  # --> permutation is fixed but differs at each step
```

Comment:
Maybe the line was intended to get the same sequence of random numbers after a "weight blowup".
Since nobody knows why it's actually there, and it's also not present in the original implementation I agree on removing it.

This will also make the patch much simpler, and won't change the function's API.